src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -359,6 +390,49 @@ internal void OnCompilationEnd(CompilationAnalysisContext context)
                 }
             }
 
+            private void ProcessTypeForwardedAttributes(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
+            {
+                var typeForwardedToAttribute = compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.TypeForwardedToAttribute");
+
+                if (typeForwardedToAttribute != null)
+                {
+                    foreach (var attribute in compilation.Assembly.GetAttributes())
+                    {
+                        if (typeForwardedToAttribute.Equals(attribute.AttributeClass))
+                        {
+                            var forwardedType = attribute.ConstructorArguments.Single().Value as INamedTypeSymbol;
```

Comment:
@AlekseyTs Cleaned up the parameter handling. The symbol comparison is left as-is: this is the pattern used in analyzers and the IDE for types like this; there isn't a public API in the compiler to easily and cheaply ask for "give me the qualified name" for an INamedTypeSymbol.